src/Api/Serializer/RestSerializer.php

```diff
@@ -276,19 +329,25 @@ private function hasPayloadParam(StructureShape $input, $payload)
     {
         if ($payload) {
             $potentiallyEmptyTypes = ['blob','string'];
-            if ($this->api->getMetadata('protocol') == 'rest-xml') {
+            if ($this->api->getProtocol() == 'rest-xml') {
                 $potentiallyEmptyTypes[] = 'structure';
             }
+
             $payloadMember = $input->getMember($payload);
-            if (in_array($payloadMember['type'], $potentiallyEmptyTypes)) {
+            //unions may also be empty/unset
```

Comment:
See test case id (in `input/rest-json.json`) `RestJsonHttpPayloadWithUnsetUnion`.  This behavior seems to only apply to `rest-json` and I was unable to find clear documentation on it, as most of the Smithy docs indicate union types must have one member that is set and not null.